File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int element = 0;
        int cnt = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == element)
                cnt++;
            else if (cnt == 0) {
                element = nums[i];
                cnt = 1;
            } else
                cnt--;
        }

        return element;
    }
};

File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    count := 0;
    candidate := nums[0];

    for _, num := range nums {
        if count == 0 {
            candidate = num
        }

        if candidate == num {
            count++
        } else {
            count--
        }
    }

    return candidate;
}